airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -1374,6 +1377,62 @@ class ContactsAllBase(Stream):
     records_field = None
     filter_field = None
     filter_value = None
+    _state = {}
+    limit_field = "count"
+    limit = 100
+
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        return self._state
+
+    @state.setter
+    def state(self, value: MutableMapping[str, Any]) -> None:
+        self._state = value
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        This is a specialized read_records for resumable full refresh that only attempts to read a single page of records
+        at a time and updates the state w/ a synthetic cursor based on the Hubspot cursor pagination value `vidOffset`
+        """
+
+        next_page_token = stream_slice
+        try:
+            properties = self._property_wrapper
```

Comment:
I fear this part as it sounds like it is the same logic as `Stream.read_records` and if one change, we now need to know that we need to change the other. Could we have the logic still centralize in `Stream`, have the `Stream._state` and only implement the `state` getter and setter in `ContactsAllBase`? Even then, I think we will need another class as [this](https://github.com/airbytehq/airbyte-internal-issues/issues/7674) is in progress and will rely on `ContactsAllBase` but I don't think it'll be RFR compatible as-is with both client side filtering incremental and RFR incremental...